integration/standalone/src/app.ts

```diff
@@ -65,15 +77,52 @@ async function initialize(client: GLSPClient): Promise<void> {
     )
     .then(() => dispatchAfterModelInitialized(actionDispatcher));
   actionDispatcher.dispatch(new RequestTypeHintsAction(diagramType));
-  actionDispatcher.dispatch(new EnableToolPaletteAction());
+  if (isInViewerMode() || isInPreviewMode()) {
+    setViewerMode();
+  } else {
+    actionDispatcher.dispatch(new EnableToolPaletteAction());
+  }
+  if (!isInPreviewMode()) {
+    actionDispatcher.dispatch(new EnableViewportAction());
+  }
 }
 
-function dispatchAfterModelInitialized(dispatcher: IActionDispatcher): void {
-  if (dispatcher instanceof GLSPActionDispatcher && selectElementIds) {
-    dispatcher
-      .onceModelInitialized()
-      .finally(() => dispatcher.dispatch(new SelectAction(selectElementIds.split(NavigationTarget.ELEMENT_IDS_SEPARATOR))));
+function dispatchAfterModelInitialized(dispatcher: GLSPActionDispatcher): void {
+  if (isNumeric(zoom)) {
+    addActions(new IvySetViewportZoomAction(+zoom / 100));
+    if (highlight) {
+      addActions(new CenterAction([highlight], false, true));
```

Comment:
I think you misunderstood my point...

I definitely would use the action variable here as a local variable and also this addActions is not needed IMO.
What I don't like are the if's here. You check e.g. twice if the highlight is set and do more or less the same. So instead of this do something like:

```java
if (isNumeric(zoom)) {
  actions.push(new IvySetViewportZoomAction());
  actions.push(showElement(CenterAction));
} else {
  actions.push(showElement(MoveIntoViewportAction));
}

function showElement(action: Action): Action[] {
  -> select action
  -> center it or move it into viewport
}
```